formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/services/FormstackService.scala

```diff
@@ -14,16 +14,14 @@ import scalaj.http.{BaseHttp, Http, HttpResponse}
 trait FormstackRequestService {
   def accountFormsForGivenPage(page: Int, accountToken: FormstackAccountToken): Either[Throwable, FormsResponse]
   def formSubmissionsForGivenPage(page: Int, formId: String, minTime: SubmissionTableUpdateDate, encryptionPassword: String, accountToken: FormstackAccountToken): Either[Throwable, FormSubmissions]
-  def submissionData(submissionIdEmails: List[SubmissionIdEmail], config: PerformLambdaConfig): Either[Throwable, List[FormstackSubmissionQuestionAnswer]]
-  def deleteUserData(submissionIdEmails: List[SubmissionIdEmail], config: PerformLambdaConfig): Either[Throwable, List[SubmissionDeletionReponse]]
+  def submissionData(requestEmail:String, submissionIdEmails: List[SubmissionIdEmail], config: PerformLambdaConfig): Either[Throwable, List[FormstackSubmissionQuestionAnswer]]
+  def deleteUserData(requestEmail:String, submissionIdEmails: List[SubmissionIdEmail], config: PerformLambdaConfig): Either[Throwable, List[SubmissionDeletionReponse]]
 }
-
-case class SubmissionsResponse(found: List[Submission], notFound: List[SubmissionIdEmail])
-sealed trait getSubmissionResult
-case class Found(submission:Submission) extends getSubmissionResult
-case class Skipped(submissionIdEmail: SubmissionIdEmail) extends getSubmissionResult
-case class AccountQuestionsAnswersResult(found: List[FormstackSubmissionQuestionAnswer], notFound: List[SubmissionIdEmail] )
-case class ValidatedSubmissions(accountOneResponse:SubmissionsResponse, accountTwoResponse: SubmissionsResponse)
+//this used to group results to multiple formstack calls in found and not found results.
+//for not found results we basically have just the id and for the found results we have whatever the particular api call would return
+case class FormstackResponses[T](found:List[T], notFound: List[SubmissionIdEmail])
+//this just used to group results of fetching submissions by which account they were fetched from. This might not agree with the account number dynamodb might assign to the submission
+case class ValidatedSubmissions(accountOneResponse:FormstackResponses[Submission], accountTwoResponse: FormstackResponses[Submission])
```

Comment:
I cleared out some of these classes. The `getSubmissionResult` I ended up removing in favour of just using either as it was only used inside a single function one time.
The two case classes that basically separate results in `found` and `not found` I just refactored them into a single generic class